pkg/controller/operators/catalog/operator.go

```diff
@@ -962,31 +962,34 @@ func (o *Operator) syncResolvingNamespace(obj interface{}) error {
 		// not-satisfiable error
 		if _, ok := err.(solver.NotSatisfiable); ok {
 			logger.WithError(err).Debug("resolution failed")
-			subs = o.setSubsCond(subs, v1alpha1.SubscriptionResolutionFailed, "ConstraintsNotSatisfiable", err.Error(), true)
-			_, updateErr := o.updateSubscriptionStatuses(subs)
+			_, updateErr := o.updateSubscriptionStatuses(
+				o.setSubsCond(subs, v1alpha1.SubscriptionCondition{
```

Comment:
suggestion: Maybe factor the reason strings out as constants? Another suggestion might be to create a couple of factory methods for creating the `SubscriptionCondition`s. Maybe it just takes the message as input? e.g.
```
func NewResolutionFailedCondition(msg string) *v1alpha1.SubscriptionCondition
```

Reply:
I think this is also a reasonable suggestion - tackle this in a follow-up for a general refactoring of this method?